tools/azsdk-cli/docs/specs/0-scenario-2.spec.md

```diff
@@ -0,0 +1,818 @@
+
+### 3. TypeSpec Customizations
+
+**Command:**
+
+```bash
+azsdk customize typespec --service healthdataaiservices --spec-path <path>
+```
+
+**Options:**
+
+- `--service <name>`: Target service short name (required)
+- `--spec-path <path>`: Path to the local `azure-rest-api-specs` clone (required)
+- `--language <code>`: Optional language filter for validation
+- `--validate-only`: Run validation without persisting updated customizations
+
+*CLI viability note: Direct TypeSpec customization via CLI may be less discoverable than agent-guided edits. If kept, prefer a minimal diff + validation summary or a `--json` flag. Otherwise defer complex refactors to Agent Mode.*
+
+### 4. Code Customizations
+
+**Command:**
+
+```bash
+azsdk customized-code update --service healthdataaiservices --languages Dotnet Java JavaScript Python Go
+```
+
+**Options:**
+
+- `--service <name>`: Service short name (required)
+- `--languages <list>`: Space-separated list of languages to update (default: all)
+- `--apply-templates`: Rehydrate customization templates with the latest generated files
+- `--dry-run`: Display planned changes without writing to disk
+
+*CLI viability note: Applying handwritten layers is often iterative and review-heavy; Agent Mode may be more appropriate. If retained for CLI Mode, focus on a dry-run diff (`--dry-run`) and a structured list of changed files per language for auditability.*
+
+### 5. Manage Test Resources
+
+**Command:**
+
+```bash
+azsdk test-resources manage --service healthdataaiservices --action create
+```
+
+**Options:**
+
+- `--service <name>`: Service short name (required)
+- `--action <create|delete>`: Action to perform (required)
+- `--resource-group <name>`: Override the default resource group naming convention
+- `--subscription <id>`: Target a specific Azure subscription
+- `--location <region>`: Azure region for resource deployment (default: westus2)
+
+**Expected Output:**
+
+```text
+Provisioning live-test resources for healthdataaiservices...
+
+✓ Resource group azsdk-healthdataaiservices-tests created
+✓ Key Vault configured
+✓ Storage account created
+✓ App Configuration deployed
+
+Test resources ready for live testing.
+```
+
+### 6. Create Test Assets
+
+**Command:**
+
+```bash
+azsdk test-assets create --service healthdataaiservices --languages Dotnet Java JavaScript Python Go
+```
+
+**Options:**
+
+- `--service <name>`: Service short name (required)
+- `--languages <list>`: Space-separated list of languages (default: all)
+- `--output-path <path>`: Override default test assets location
+
+**Expected Output:**
+
+```text
+Creating test asset files for healthdataaiservices...
+
+✓ .NET: test-resources.json created
+✓ Java: test-resources.json created
+✓ JavaScript: test-resources.json created
+✓ Python: test-resources.json created
+✓ Go: test-resources.json created
+
+Test assets created for 5/5 languages.
+```
+
+### 7. Set Test Mode
+
+**Command:**
+
+```bash
+azsdk test-mode set --mode live-record
```

Comment:
This probably wouldn't be an actual command and would instead be an option that we pass into each test run.